File: ReverseArray.cpp
#include <iostream>
using namespace std;
void printArray(int arr[],int size)
{
    cout<<"Print your Array: "<<endl;
    for(int i=0;i<size;i++)
    {
        cout<<arr[i]<<" ";
    }
    cout<<endl;
}
void swap1(int *val1,int *val2)
{
    int temp;
    temp=*val1;
    *val1=*val2;
    *val2=temp;
}
void reverseArray(int arr[],int size)
{
    int start=0;
    int end=size-1;
    while(start<=end)
    {
        swap1(&arr[start],&arr[end]);
        start++;
        end--;
    }
    cout<<"reverse";
}
int main()
{
    int array1[]={1,2,3,4,5};
    int array2[]={11,12,13,14};
    reverseArray(array1,5);
    reverseArray(array2,4);
    printArray(array1,5);
    printArray(array2,4);
    return 0;
}

File: ReverseArray.java
import java.util.Arrays;

public class ReverseArray {
    public static void main(String[] args) {
        int[] arr=new int[]{1,2,3,4,5};
        reverseArrayUsingRecursion(arr,0,arr.length-1);
        System.out.println(Arrays.toString(arr));
    }
    public static void reverseArrayUsingRecursion(int[] arr,int start,int end)
    {
        if(start>end) return;

        int temp=arr[start];
        arr[start]=arr[end];
        arr[end]=temp;

        reverseArrayUsingRecursion(arr,start+1,end-1);
    }
}
